app/services/prisoner_location.rb

```diff
@@ -12,6 +12,10 @@ def initialize(offender, prison_code = nil)
     self.prison_code = prison_code
   end
 
+  def internal_location
+    prisoner_location.internal_location if valid?
```

Comment:
This object now gets `valid?` called multiple times in its lifecycle for no need. When we build the object we could call `valid?` on it and remove the other calls on it.

Reply:
Also, is the check needed here? When the object is invalid it returns `nil`, which is the same value that an instance of `Nomis::Establishment` would return.

Reply:
I don't see much problem with an object asking itself about its internal state.

So here we do need to ask if it is valid or not because the prisoner might not be located at the prison provided by the visitor.
I'm going to rename this object to `PrisonerLocationValidation` to make the intent of the object a bit clearer.